sklearn/cluster/tests/test_optics.py

```diff
@@ -209,6 +210,40 @@ def test_bad_reachability():
         clust.fit(X)
 
 
+def test_nowarn_if_metric_bool_data_bool():
+    pairwise_metric = 'rogerstanimoto'
+    X = np.random.randint(2, size=(5, 2), dtype=np.bool)
+
+    with pytest.warns(None) as warn_record:
+        OPTICS(metric=pairwise_metric).fit(X)
+        assert len(warn_record) == 0
+
+
+def test_warn_if_metric_bool_data_no_bool():
+    pairwise_metric = 'rogerstanimoto'
+    X = np.random.randint(2, size=(5, 2), dtype=np.int)
+    msg = f"Data will be converted to boolean for" \
```

Comment:
same here: use parenthesis instead of backslashes please
(note: you don't need to catch the whole message, a meaningful subpart will be fine as well)

Reply:
Thanks for your quick response and review @NicolasHug ! I have pushed the resolutions and hope that they can be merged soon.